docs/2_3_common_states.adoc

```diff
@@ -82,485 +87,644 @@ If `loggingOn == fmi3True`, the FMU enables a vendor defined set of `<LogCategor
+
+** Co-Simulation: If <<toleranceDefined,`toleranceDefined == fmi3True`>>, then the communication step size of the FMU is controlled by error estimation.
+In case the FMU utilizes a numerical integrator with variable step size and error estimation, it is suggested to use <<tolerance>> for the error estimation of the integrator (usually as relative tolerance). +
+An FMU for Co-Simulation might ignore this argument.
 
-Functions <<get-and-set-variable-values,`fmi3Get{VariableType}`>>::
-See <<get-and-set-variable-values>> for the general mechanism to get variable values.
-Getting variable might trigger computations.
-_[If *Terminated* is entered because of an `fmi3Error` return value, retrieved values should only be used for debugging purposes.]_
 
-===== State: Instantiated [[Instantiated]]
+* [[startTime,`startTime`]]`startTime` and
+* [[stopTime,`stopTime`]]`stopTime` can be used to check whether the model is valid within the given boundaries, or to allocate the necessary memory for storing results.
+<<startTime>> is the <<fixed>> <<initial>> value of the <<independent>> variable and inherits its unit.
++
+_[It is defined with <<causality>> = <<independent>> in the <<modelDescription.xml>>._
+_If the <<independent>> variable is `time`, <<startTime>> is the starting time of initialization.]_
 
-In this state the FMU can do one-time initializations and allocate memory.
-This state is entered after <<fmi3Instantiate>> returned successfully.
+* [[stopTimeDefined,`stopTimeDefined`]] If `stopTimeDefined == fmi3True`, then <<stopTime>> is the final value of the <<independent>> variable and inherits its unit.
+If the environment tries to compute past <<stopTime>>, the FMU has to return <<fmi3Error,`fmi3Status == fmi3Error`>>.
+If <<stopTimeDefined,`stopTimeDefined == fmi3False`>>, then no final value of the <<independent>> variable is defined and argument <<stopTime>> is meaningless.
 
-// TODO: Configuration Mode
-// TODO: In CS is an iteration variable i
+===== State: Initialization Mode [[InitializationMode,*Initialization Mode*]]
+
+The <<InitializationMode>> is used by the simulation algorithm to compute consistent initial conditions for the overall system.
+Equations are active to determine the initial FMU state, as well as all <<output,`outputs`>> (and optionally other variables exposed by the exporting tool).
+Artificial or real algebraic loops over connected FMUs in <<InitializationMode>> may be handled by using appropriate numerical algorithms.
 
-Mathematical Description::
+In <<InitializationMode>>, the FMU computes initial values at the start time latexmath:[\mathbf{t}_\mathit{start}] using function latexmath:[\mathbf{f}_\mathit{init}], not present in the other modes, for example, equations to define the <<start>> value for a <<state>> or for the derivative of a <<state>>.
```

Comment:
I have trouble understanding this sentence. Are we defining what f_init is? Why is it defined as a function? During initialization mode, the FMU activates a set of equations. We can call that set f_init, but I wouldn't call it a function...

Reply:
We support iteration for solving the initial value problem. For that, those equations have to be computed.